jmx-scraper/src/main/java/io/opentelemetry/contrib/jmxscraper/JmxScraper.java

```diff
@@ -0,0 +1,138 @@
+
+  // TODO depend on instrumentation 2.9.0 snapshot
+  // private final JmxMetricInsight service;
+
+  /**
+   * Main method to create and run a {@link JmxScraper} instance.
+   *
+   * @param args - must be of the form "-config {jmx_config_path,'-'}"
+   */
+  @SuppressWarnings({"SystemOut", "SystemExitOutsideMain"})
+  public static void main(String[] args) {
+    JmxScraperConfig config;
+    JmxScraper jmxScraper = null;
+    try {
+      JmxScraperConfigFactory factory = new JmxScraperConfigFactory();
+      config = JmxScraper.createConfigFromArgs(Arrays.asList(args), factory);
+      jmxScraper = new JmxScraper(config);
+
+    } catch (ArgumentsParsingException e) {
+      System.err.println(
+          "Usage: java -jar <path_to_jmxscraper.jar> "
+              + "-config <path_to_config.properties or - for stdin>");
+      System.exit(1);
+    } catch (ConfigurationException e) {
+      System.err.println(e.getMessage());
+      System.exit(1);
+    }
+
+    try {
+      Objects.requireNonNull(jmxScraper).start();
+    } catch (IOException e) {
+      System.err.println("Unable to connect " + e.getMessage());
+      System.exit(2);
+    }
+  }
+
+  /**
+   * Create {@link JmxScraperConfig} object basing on command line options
+   *
+   * @param args application commandline arguments
+   */
+  static JmxScraperConfig createConfigFromArgs(List<String> args, JmxScraperConfigFactory factory)
+      throws ArgumentsParsingException, ConfigurationException {
+    if (!args.isEmpty() && (args.size() != 2 || !args.get(0).equalsIgnoreCase("-config"))) {
+      throw new ArgumentsParsingException();
+    }
+
+    Properties loadedProperties = new Properties();
+    if (args.size() == 2) {
+      String path = args.get(1);
+      if (path.trim().equals("-")) {
+        loadPropertiesFromStdin(loadedProperties);
+      } else {
+        loadPropertiesFromPath(loadedProperties, path);
+      }
+    }
+
+    return factory.createConfig(loadedProperties);
+  }
+
+  private static void loadPropertiesFromStdin(Properties props) throws ConfigurationException {
+    try (InputStream is = new DataInputStream(System.in)) {
+      props.load(is);
+    } catch (IOException e) {
+      throw new ConfigurationException("Failed to read config properties from stdin", e);
+    }
+  }
+
+  private static void loadPropertiesFromPath(Properties props, String path)
+      throws ConfigurationException {
+    try (InputStream is = Files.newInputStream(Paths.get(path))) {
+      props.load(is);
+    } catch (IOException e) {
+      throw new ConfigurationException("Failed to read config properties file: '" + path + "'", e);
+    }
+  }
+
+  JmxScraper(JmxScraperConfig config) throws ConfigurationException {
+    String serviceUrl = config.getServiceUrl();
+    int interval = config.getIntervalMilliseconds();
+    if (interval < 0) {
+      throw new ConfigurationException("interval must be positive");
+    }
+    this.client = JmxRemoteClient.createNew(serviceUrl);
```

Comment:
nitpick: constructors shouldn't do work. This constructor is doing both validation and creation of true dependencies (violation of DI principles). This can be worked around with a static factory method.

It's also odd to me that the interval is verified here, but neither the config nor the interval are passed to the client. That feels wrong.